crates/sui-keys/src/random_names.rs

```diff
@@ -0,0 +1,199 @@
+        "focused",
+        "friendly",
+        "frosty",
+        "funny",
+        "gallant",
+        "gifted",
+        "goofy",
+        "gracious",
+        "great",
+        "happy",
+        "hardcore",
+        "heuristic",
+        "hopeful",
+        "hungry",
+        "infallible",
+        "inspiring",
+        "intelligent",
+        "interesting",
+        "jolly",
+        "jovial",
+        "keen",
+        "kind",
+        "laughing",
+        "loving",
+        "lucid",
+        "magical",
+        "modest",
+        "musing",
+        "mystifying",
+        "naughty",
+        "nervous",
+        "nice",
+        "nifty",
+        "nostalgic",
+        "objective",
+        "optimistic",
+        "peaceful",
+        "pedantic",
+        "pensive",
+        "practical",
+        "priceless",
+        "quirky",
+        "quizzical",
+        "recursing",
+        "relaxed",
+        "reverent",
+        "romantic",
+        "sad",
+        "serene",
+        "sharp",
+        "silly",
+        "sleepy",
+        "stoic",
+        "strange",
+        "stupefied",
+        "suspicious",
+        "sweet",
+        "tender",
+        "thirsty",
+        "trusting",
+        "unruffled",
+        "upbeat",
+        "vibrant",
+        "vigilant",
+        "vigorous",
+        "wizardly",
+        "wonderful",
+        "xenodochial",
+        "youthful",
+        "zealous",
+        "zen",
+    ];
+
+    let right_names = vec![
+        "agates",
+        "alexandrite",
+        "amber",
+        "amethyst",
+        "apatite",
+        "avanturine",
+        "axinite",
+        "beryl",
+        "beryl",
+        "carnelian",
+        "chalcedony",
+        "chrysoberyl",
+        "chrysolite",
+        "chrysoprase",
+        "coral",
+        "corundum",
+        "crocidolite",
+        "cyanite",
+        "cymophane",
+        "diamond",
+        "dichroite",
+        "emerald",
+        "epidote",
+        "euclase",
+        "felspar",
+        "garnet",
+        "heliotrope",
+        "hematite",
+        "hiddenite",
+        "hypersthene",
+        "idocrase",
+        "jasper",
+        "jet",
+        "labradorite",
+        "malachite",
+        "moonstone",
+        "obsidian",
+        "opal",
+        "pearl",
+        "phenacite",
+        "plasma",
+        "prase",
+        "quartz",
+        "ruby",
+        "sapphire",
+        "sphene",
+        "spinel",
+        "spodumene",
+        "sunstone",
+        "topaz",
+        "tourmaline",
+        "turquois",
+        "zircon",
+    ];
+
+    let mut rng = thread_rng();
+    let mut output = generate(&mut rng, &left_names, &right_names);
+    // as long as the generated name is in the list of conflicts,
+    // we try to find a different name that is not in the list yet
+    loop {
+        if conflicts.contains(&output) {
+            output = generate(&mut rng, &left_names, &right_names);
+        } else {
+            break;
+        }
+    }
+    output
```

Comment:
```suggestion
    loop {
        let output = generate(&mut rng, &left_names, &right_names);
        if !conflicts.contains(&output) {
            return output;
        }
    }
```

Reply:
🤦 obviously that's a lot nicer :D I'll fix it.